CDP4Reporting/DataCollection/DataCollectorNode.cs

```diff
@@ -389,59 +411,120 @@ internal void AddDataRows(DataTable table, bool excludeMissingParameters)
         /// </returns>
         private void AddDataRow(DataTable table)
         {
-            var row = table.NewRow();
-            table.Rows.Add(row);
-
-            var rowPresentation = this.GetRowRepresentation();
-
-            this.InitializeCategoryColumns(row);
+            var rowPresentationTuples = this.FillStateDependentParameterColumns(table);
 
-            foreach (var rowField in this.normalColumns)
+            if (!rowPresentationTuples.Any())
             {
-                var column = rowField.Key.GetValue(rowPresentation) as DataCollectorColumn<T>;
-
-                column?.Populate(table, row);
+                rowPresentationTuples.Add((this.GetRowRepresentation(), table.NewRow()));
             }
 
-            foreach (var publicGetter in this.publicGetterProperties)
+            foreach (var (rowPresentation, dataRow) in rowPresentationTuples)
             {
-                row[publicGetter.Name] = publicGetter.GetMethod.Invoke(
-                    this.GetRowRepresentation(),
-                    new object[] { });
+                table.Rows.Add(dataRow);
+
+                this.FillCategoryDecompositionHierarchyColumns(dataRow);
+                this.FillNormalColumns(rowPresentation, dataRow);
+                this.FillPublicGetterColumns(rowPresentation, dataRow);
             }
+        }
 
-            var removeOriginalColumn = false;
+        /// <summary>
+        /// Fills the <see cref="DataColumn"/>s with data for properties of an instance of <see cref="T"/> whose <see cref="Type"/> implement <see cref="IDataCollectorStateDependentPerRowParameter"/>.
```

Comment:
This line is a bit too long, I would suggest splitting it before "implement <see ...".